engine/src/main/java/io/zeebe/engine/processing/bpmn/WorkflowInstanceStateTransitionGuard.java

```diff
@@ -115,19 +122,31 @@ public boolean isValidStateTransition(final BpmnElementContext context) {
   }
 
   private Either<String, ElementInstance> hasElementInstanceInState(
-      final ElementInstance elementInstance, final WorkflowInstanceIntent expectedState) {
+      final ElementInstance elementInstance,
+      final WorkflowInstanceIntent expectedState,
+      final WorkflowInstanceIntent... otherExpected) {
     final var currentState = elementInstance.getState();
-    if (currentState != expectedState) {
+    if (expectedState != currentState || !Arrays.asList(otherExpected).contains(currentState)) {
       return Either.left(
           String.format(
-              "Expected element instance to be in state '%s' but was '%s'.",
-              expectedState, currentState));
-
+              "Expected element instance to be in state '%s' or one of '%s' but was '%s'.",
+              expectedState, Arrays.toString(otherExpected), currentState));
     } else {
       return Either.right(elementInstance);
     }
   }
 
+  private Either<String, ?> hasElementInstanceWithState(
```

Comment:
Do we need both methods? (`hasElementInstanceWithState/inState`) If one is supposed to be a shortcut to call the other, maybe we can always use the shortcut and inline the called method

Reply:
I've tried this, but it looks worse to me.